### ISSUE TITLE
Event of Death

### ISSUE THREAD

**Issue by dpc** [urbit/urbit#1271]
My planet received an EoD (event of death). Keeps processing, slowly increasing memory usage. `kill -9` took 10 minutes to kill it the first time I've noticed. Restarting get stuck on trying to replay this event.

```
---------------- playback starting----------------
rest: replaying events 86.964 through 86.964
```

 Is this useful/interesting?

**Comment by vvisigoth** [comment]
Very interesting.

@joemfb
—
~poldec-tonteg
http://urbit.org

**Comment by dpc** [comment]
Correction: the memory does not seem to increase now. I definitely had to wait a long while after issuing `kill -9` so I assumed it has something to do with swap. But I am not sure and now when I restart Urbit I can kill it just fine instantly. It is just consuming 100% CPU all the time.

**Comment by joemfb** [comment]
@dpc, this would have had me complete stumped, but we actually saw the exact same symptoms on a star today. If this is the same bug, the cause is corruption at the end of the event log causing an infinite loop while reading events for replay.

We were able to fix the star by trimming the corrupt data off of the end of the log. This is wandering off the beaten path of course, but if you'd like to try, @ixv can fill you in on the exact details (it involved adding printfs to the replay to identify the position of the last good event). It should also work to trim a byte at time and attempt replay. You'd know that the event has finished running by the `.` that gets printed.

As for the bug itself, we are putting the finishing touches on a branch (cc-release) with an entirely new event-log (along with a bunch of other things), so I don't think we'll put the work into fixing this older implementation, which is known to be have these kinds of issues. My apologies all the same.
